test/e2e/common/runtime.go

```diff
@@ -248,144 +255,109 @@ while true; do sleep 1; done
-							reason := status.State.Waiting.Reason
-							if reason != images.ErrImagePull.Error() &&
-								reason != images.ErrImagePullBackOff.Error() {
-								return fmt.Errorf("unexpected waiting reason: %q", reason)
-							}
-						}
-						// Check pod phase
-						phase, err := container.GetPhase()
-						if err != nil {
-							return fmt.Errorf("failed to get pod phase: %v", err)
+					}
+					if expectedPullStatus {
+						if status.State.Waiting == nil {
+							return fmt.Errorf("expected container state: Waiting, got: %q",
+								GetContainerState(status.State))
 						}
-						if phase != testCase.phase {
-							return fmt.Errorf("expected pod phase: %q, got: %q", testCase.phase, phase)
+						reason := status.State.Waiting.Reason
+						if reason != images.ErrImagePull.Error() &&
+							reason != images.ErrImagePullBackOff.Error() {
+							return fmt.Errorf("unexpected waiting reason: %q", reason)
 						}
-						return nil
 					}
-					// The image registry is not stable, which sometimes causes the test to fail. Add retry mechanism to make this
-					// less flaky.
-					const flakeRetry = 3
-					for i := 1; i <= flakeRetry; i++ {
-						var err error
-						By("create the container")
-						container.Create()
-						By("check the container status")
-						for start := time.Now(); time.Since(start) < ContainerStatusRetryTimeout; time.Sleep(ContainerStatusPollInterval) {
-							if err = checkContainerStatus(); err == nil {
-								break
-							}
-						}
-						By("delete the container")
-						container.Delete()
-						if err == nil {
+					// Check pod phase
+					phase, err := container.GetPhase()
+					if err != nil {
+						return fmt.Errorf("failed to get pod phase: %v", err)
+					}
+					if phase != expectedPhase {
+						return fmt.Errorf("expected pod phase: %q, got: %q", expectedPhase, phase)
+					}
+					return nil
+				}
+
+				// The image registry is not stable, which sometimes causes the test to fail. Add retry mechanism to make this less flaky.
+				const flakeRetry = 3
+				for i := 1; i <= flakeRetry; i++ {
+					var err error
+					By("create the container")
+					container.Create()
+					By("check the container status")
+					for start := time.Now(); time.Since(start) < ContainerStatusRetryTimeout; time.Sleep(ContainerStatusPollInterval) {
+						if err = checkContainerStatus(); err == nil {
 							break
 						}
-						if i < flakeRetry {
-							framework.Logf("No.%d attempt failed: %v, retrying...", i, err)
-						} else {
-							framework.Failf("All %d attempts failed: %v", flakeRetry, err)
-						}
 					}
-				})
+					By("delete the container")
+					container.Delete()
+					if err == nil {
+						break
+					}
+					if i < flakeRetry {
+						framework.Logf("No.%d attempt failed: %v, retrying...", i, err)
+					} else {
+						framework.Failf("All %d attempts failed: %v", flakeRetry, err)
+					}
+				}
 			}
+
+			It("should not be able to pull image from invalid registry", func() {
```

Comment:
Is there a reason for splitting out the grid tests into individual tests?  I think I like it, but I'm wondering if an actual decision was made somewhere that is motivating this...

Reply:
Yeah, actually I try to convert these to conformance tests. But walker.go isn't reading the test case string from grid test. So splitting out the grid test.

Reply:
I found similar PR of runtime.go. [https://github.com/kubernetes/kubernetes/pull/70492](url)
Could you tell me the relation?

Reply:
@k-toyoda-pi  this issue came from that PR, this PR is working in progress. And this PR to fix only inconsistent message.

Reply:
Thanks for reply. I understand you are investigating about "\n" with the PR.